File: sdk/clojure/src/hitechcloud/sdk/contacts.clj
(ns hitechcloud.sdk.contacts
  (:require [hitechcloud.sdk.http-client :as http]))

;; Contacts: manage account contacts

(defn list "GET /api/contact - Return a list of contacts on this account" [client & [params]] (http-get client "/api/contact" params))

(defn create "POST /api/contact - Create new contact account" [client & [params]] (http-post client "/api/contact" params))

(defn get_privileges "GET /api/contact/privileges - List possible contact privileges" [client & [params]] (http-get client "/api/contact/privileges" params))

(defn get "GET /api/contact/{id} - Get contact details" [client & [params]] (http-get client "/api/contact/{id}" params))

(defn update "PUT /api/contact/{id} - Update contact details" [client & [params]] (http-put client "/api/contact/{id}" params))

File: sdk/fsharp/src/HiTechCloudSDK/Contacts.fs
namespace HiTechCloud.SDK

/// Contacts: manage account contacts
type ContactsResource(httpClient: HttpClient) =

    /// GET /api/contact - Return a list of contacts on this account
    member this.List() =
        async {
            return! httpClient.getAsync("/api/contact")
        }

    /// POST /api/contact - Create new contact account
    member this.Create() =
        async {
            return! httpClient.postAsync("/api/contact")
        }

    /// GET /api/contact/privileges - List possible contact privileges
    member this.GetPrivileges() =
        async {
            return! httpClient.getAsync("/api/contact/privileges")
        }

    /// GET /api/contact/{id} - Get contact details
    member this.Get() =
        async {
            return! httpClient.getAsync("/api/contact/{id}")
        }

    /// PUT /api/contact/{id} - Update contact details
    member this.Update() =
        async {
            return! httpClient.putAsync("/api/contact/{id}")
        }

